File: backend/semgrep_rules/tests/java/kisa_sw_06_dangerous_file_upload.java
import java.io.File;
import org.springframework.web.multipart.MultipartFile;

class kisa_sw_06_dangerous_file_upload {
    void unsafe(MultipartFile upload) throws Exception {
        // ruleid: kisa.sw06.java.unvalidated-upload-filename
        upload.transferTo(new File(upload.getOriginalFilename()));
    }

    void safe(MultipartFile upload, File generated) throws Exception {
        // ok: kisa.sw06.java.unvalidated-upload-filename
        upload.transferTo(generated);
    }
}

File: backend/semgrep_rules/tests/javascript/kisa_sw_06_dangerous_file_upload.js
const fs = require("fs");

function unsafe(file, data) {
  // ruleid: kisa.sw06.javascript.unvalidated-upload-filename
  fs.writeFileSync(file.originalname, data);
}

function safe(file, generatedPath, data) {
  // ok: kisa.sw06.javascript.unvalidated-upload-filename
  fs.writeFileSync(generatedPath, data);
}
